src/chordSymbol.js

```diff
@@ -0,0 +1,593 @@
+    return this.vertical;
+  }
+
+  // Get and set horizontal justification. `justification` is a value in
+  // `Annotation.Justify`.
+  setHorizontal(just) {
+    this.horizontal = typeof (just) === 'string'
+      ? ChordSymbol.HorizontalJustifyString[just]
+      : just;
+    return this;
+  }
+
+  getWidth() {
+    let rv = 0;
+    this.symbolBlocks.forEach((symbol) => {
+      rv += symbol.width;
+    });
+    return rv;
+  }
+
+  isSuperscript(symbol) {
+    return symbol.symbolModifier !== null && symbol.symbolModifier === ChordSymbol.SymbolModifiers.SUPERSCRIPT;
+  }
+
+  isSubscript(symbol) {
+    return symbol.symbolModifier !== null && symbol.symbolModifier === ChordSymbol.SymbolModifiers.SUBSCRIPT;
+  }
+
+  // Render text and glyphs above/below the note
+  draw() {
+    this.checkContext();
+
+    if (!this.note) {
+      throw new Vex.RERR(
+        'NoNoteForAnnotation', "Can't draw text annotation without an attached note."
+      );
+    }
+
+    this.setRendered();
+    const start = this.note.getModifierStartXY(Modifier.Position.ABOVE,
+      this.index);
+
+    // We're changing context parameters. Save current state.
+    this.context.save();
+    this.context.setFont(this.font.family, this.font.size, this.font.weight);
+
+    let y;
+
+    let stem_ext;
+    let spacing;
+    const has_stem = this.note.hasStem();
+    const stave = this.note.getStave();
+
+    // The position of the text varies based on whether or not the note
+    // has a stem.
+    if (has_stem) {
+      stem_ext = this.note.getStem().getExtents();
+      spacing = stave.getSpacingBetweenLines();
+    }
+
+    if (this.vertical === ChordSymbol.VerticalJustify.BOTTOM) {
+      // HACK: We need to compensate for the text's height since its origin
+      // is bottom-right.
+      y = stave.getYForBottomText(this.text_line + Flow.TEXT_HEIGHT_OFFSET_HACK);
+      if (has_stem) {
+        const stem_base = (this.note.getStemDirection() === 1 ? stem_ext.baseY : stem_ext.topY);
+        y = Math.max(y, stem_base + (spacing * (this.text_line + 2)));
+      }
+    } else  { // (this.vertical === ChordSymbol.VerticalJustify.TOP)
+      y = Math.min(stave.getYForTopText(this.text_line), this.note.getYs()[0] - 10);
+      if (has_stem) {
+        y = Math.min(y, (stem_ext.topY - 5) - (spacing * this.text_line));
+      }
+    }
+
+    let x = start.x;
+    if (this.horizontal === ChordSymbol.HorizontalJustify.LEFT) {
+      x = start.x;
+    } else if (this.horizontal === ChordSymbol.HorizontalJustify.RIGHT) {
+      x = start.x - this.getWidth();
+    } else if (this.horizontal === ChordSymbol.HorizontalJustify.CENTER) {
+      x = start.x - this.getWidth() / 2;
+    } else /* CENTER_STEM */ {
+      x = this.note.getStemX() -  this.getWidth() / 2;
+    }
+    this.symbolBlocks.forEach((symbol) => {
+      const sp = this.isSuperscript(symbol);
+      const sub = this.isSubscript(symbol);
+      let curY = y;
+
+      if (sp) {
+        curY -= 8;
```

Comment:
These constants should be in metrics files. (Also, how does this remain stable as the font sizes change?)

Reply:
fonts come in sizes, i forgot that.  I added some test cases for it and adjust for font size, to some effect (but it still drifts as fonts get super large/small).  I may have too many test cases now.

Reply:
![image](https://user-images.githubusercontent.com/5438280/81134751-51f32680-8f1b-11ea-8682-4482b64f3d58.png)